stdlib/LinearAlgebra/src/uniformscaling.jl

```diff
@@ -452,9 +454,9 @@ function _hvcat(rows::Tuple{Vararg{Int}}, A::Union{AbstractVecOrMat,UniformScali
     j = 0
     for i = 1:nr # infer UniformScaling sizes from row counts, if possible:
         ni = -1 # number of rows in this block-row, -1 indicates unknown
-        for k = 1:rows[i]
-            if !isa(A[j+k], UniformScaling)
-                na = size(A[j+k], 1)
+        sizes = map(a -> szfun(a, 1), A[j+1:j+rows[i]])
```

Comment:
I think this might still be unstable? You're accessing the tuple `A` at dynamic indices (unless this all ends up unrolled). You might need to compute the sizes outside of these loops and then just index the computed `sizes` at the dynamic indices.

Since this `szfun` exists at module scope (rather than as an inner function), we could consider a more instructive name like `_catsize` so that it's clear it's internal and used for concatenation.

Reply:
(I couldnt figure out how to add a line comment so far from your changes)

It looks like there's currently still some `A[...]` accesses around ln481 that should be able to use the new variables instead. Of course, check that any of this helps in the simpler `vcat` or `hcat` cases first or else none of my suggestions matter.